src/docker/remote.rs

```diff
@@ -1066,9 +1038,18 @@ pub(crate) fn run(
     .wrap_err("when copying project")?;
     let sysroot = dirs.get_sysroot().to_owned();
     let mut copied = vec![
-        (&dirs.xargo, mount_prefix_path.join("xargo")),
-        (&dirs.cargo, mount_prefix_path.join("cargo")),
-        (&sysroot, mount_prefix_path.join("rust")),
+        (
+            &dirs.xargo,
+            mount_prefix_path.join(Path::new(&dirs.xargo_mount_path_relative()?)),
+        ),
+        (
+            &dirs.cargo,
+            mount_prefix_path.join(Path::new(&dirs.cargo_mount_path_relative()?)),
+        ),
+        (
+            &sysroot,
+            mount_prefix_path.join(Path::new(&dirs.sysroot_mount_path_relative()?)),
```

Comment:
`MountFinder::join` takes an `AsRef<Path>`, the manual `Path::new` shouldn't be needed.